contracts/SwapNFT.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {ISwapNFT} from "./interfaces/ISwapNFT.sol";
+
+contract SwapNFT is ERC721, ISwapNFT {
+    address public immutable aelin;
+
+    uint256 public tokenCount;
+
+    mapping(address => bool) public hasMinted;
+
+    constructor(address _aelinAddress) ERC721("AelinSwapNFT", "SwapNFT") {
+        aelin = _aelinAddress;
+    }
+
+    function mint() external {
+        if (IERC20(aelin).balanceOf(msg.sender) == 0) revert BalanceIsZero();
```

Comment:
we can mint them a NFT even if their balance is 0. they can get the AELIN later. I am ok with keepign this check though if you want

Reply:
it's actually a good point. I will remove this line